cmd/dbaas_show_opensearch.go

```diff
@@ -0,0 +1,302 @@
+		}
+	}
+	et.Render()
+	t.Append([]string{"IndexPatterns", buf.String()})
+
+	var indexTemplate string
+	if o.IndexTemplate != nil {
+		indexTemplate = fmt.Sprintf("MappingNestedObjectsLimit:%d NumberOfReplicas:%d NumberOfShards:%d",
+			o.IndexTemplate.MappingNestedObjectsLimit,
+			o.IndexTemplate.NumberOfReplicas,
+			o.IndexTemplate.NumberOfShards)
+	}
+	t.Append([]string{"IndexTemplate", indexTemplate})
+
+	t.Append([]string{"KeepIndexRefreshInterval", fmt.Sprint(o.KeepIndexRefreshInterval)})
+	t.Append([]string{"MaxIndexCount", strconv.FormatInt(o.MaxIndexCount, 10)})
+
+	var dashboard string
+	if o.Dashboard != nil {
+		dashboard = fmt.Sprintf("Enabled:%v MaxOldSpaceSize:%d OpensearchRequestTimeout:%d",
+			o.Dashboard.Enabled, o.Dashboard.MaxOldSpaceSize, o.Dashboard.OpensearchRequestTimeout)
+	}
+	t.Append([]string{"Dashboard", dashboard})
+
+	var users string
+	for _, u := range o.Users {
+		users += fmt.Sprintf("%s (%s)\n", u.Username, u.Type)
+	}
+	t.Append([]string{"Users", users})
+}
+
+func (c *dbaasServiceShowCmd) showDatabaseServiceOpensearch(ctx context.Context) (outputter, error) {
+	res, err := cs.GetDbaasServiceOpensearchWithResponse(ctx, oapi.DbaasServiceName(c.Name))
+	if err != nil {
+		return nil, err
+	}
+	if res.StatusCode() != http.StatusOK {
+		return nil, fmt.Errorf("API request error: unexpected status %s", res.Status())
+	}
+
+	switch {
+	case c.ShowBackups:
+		return opensearchShowBackups(res.JSON200)
+	case c.ShowNotifications:
+		return opensearchShowNotifications(res.JSON200)
+	case c.ShowSettings != "":
+		return nil, opensearchShowSettings(c.ShowSettings, res.JSON200)
+	case c.ShowURI:
+		fmt.Println(defaultString(res.JSON200.Uri, ""))
+		return nil, nil
+	default:
+		return opensearchShowDatabase(res.JSON200, c.Zone)
+	}
+}
+
+func opensearchShowSettings(setting string, db *oapi.DbaasServiceOpensearch) error {
+	var serviceSettings *map[string]interface{}
+
+	switch setting {
+	case "opensearch":
+		serviceSettings = db.OpensearchSettings
+	default:
+		return fmt.Errorf("invalid settings value %q, expected one of: %s", setting, strings.Join(opensearchSettings, ", "))
+	}
+
+	if serviceSettings != nil {
+		out, err := json.MarshalIndent(serviceSettings, "", "  ")
+		if err != nil {
+			return fmt.Errorf("unable to marshal JSON: %w", err)
+		}
+		fmt.Println(string(out))
+	}
+
+	return nil
+}
+
+func opensearchShowNotifications(db *oapi.DbaasServiceOpensearch) (outputter, error) {
+	out := make(dbServiceNotificationListOutput, 0)
+	if db.Notifications != nil {
+		for _, n := range *db.Notifications {
+			out = append(out, dbServiceNotificationListItemOutput{
+				Level:   string(n.Level),
+				Message: wordwrap.WrapString(n.Message, 50),
+				Type:    string(n.Type),
+			})
+		}
+	}
+	return &out, nil
+}
+
+func opensearchShowBackups(db *oapi.DbaasServiceOpensearch) (outputter, error) {
+	out := make(dbServiceBackupListOutput, 0)
```

Comment:
It's possible to allocate exact capacity with little reordering:
```
if db.Backups == nil {
  return nil, nil //or return &dbServiceBackupListOutput{}, nil
}
out := make(dbServiceBackupListOutput, 0, len(db.Backups))
for _, b := range *db.Backups {
...
```

Reply:
Sure, it probably won't have any significant performance impact, but it looks better! (Took this part from [here](https://github.com/exoscale/cli/blob/596501c3ed402bebb406b948646d6e37c2fe9cde/cmd/dbaas_show_kafka.go#L153))